### PR TITLE
Trillian docker build

### PR DESCRIPTION
**Pull request by tommyd450**
Prior to this trillian was being built using the make file of the upstream rekor repo.

These changes remove it and use the two new docker files present in the midstream repo for trillian.
I have built and tested both images on a ROSA cluster with the sigstore/securesign stack and successfully signed and verified one of those images.

The only question I have is should these images be produced in the same git action step or should I split them?

### CODE REVIEW

**Comment by lance** [comment]
> The only question I have is should these images be produced in the same git action step or should I split them?

It's probably OK for now. We'll want to split the build and publish into two separate steps at some point, however.

File: .github/workflows/image-factory.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
         with:
           repository: securesign/fulcio
           path: fulcio
-          ref: release-next
+          ref: ${{ env.FULCIO_VER }}
```
Comment (lance) [inline .github/workflows/image-factory.yaml]: This change seems unrelated.

File: .github/workflows/image-factory.yaml [hunk 2]
```diff
@@ -169,17 +169,11 @@ jobs:
...
-          repository: securesign/rekor
-          path: rekor
-          ref: ${{ env.TRILLIAN_VER }}
-
-      - name: install ko
-        uses: imjasonh/setup-ko@v0.6
-        env:
-          KO_DOCKER_REPO: quay.io/securesign
+          repository: securesign/trillian
+          ref: release-next
```
Comment (lance) [inline .github/workflows/image-factory.yaml:176]: Once we have named/tracked branches in midstream, this should change to be something like `redhat-v1.5.2`.

**Review by lance — CHANGES_REQUESTED** [review]
Looks good, Tommy - thanks! Can you please make the one requested change?

File: .github/workflows/image-factory.yaml (hunk 2 above)
Comment (tommyd450) [reply .github/workflows/image-factory.yaml:176]: Ok so for now operating off of release-next is our m.o for the time being?

Comment (lance) [reply .github/workflows/image-factory.yaml:176]: > Ok so for now operating off of release-next is our m.o for the time being?

Yes -- but only temporarily until we get all of the midstream version tracking worked out.

File: .github/workflows/image-factory.yaml [hunk 3]
```diff
@@ -197,8 +191,14 @@ jobs:
...
-          cd rekor
-          KO_PREFIX=quay.io/securesign KO_DOCKER_REPO=quay.io/securesign KO_DEFAULTBASEIMAGE=registry.access.redhat.com/ubi9/ubi-micro make ko-trillian
+          cd trillian
+          mv redhat/overlays/log_server/Dockerfile .
+          docker build -t quay.io/securesign/trillian_log_server:${TRILLIAN_VER} .
+          docker push quay.io/securesign/trillian_log_server:${TRILLIAN_VER}
+          mv Dockerfile redhat/overlays/log_server
+          mv redhat/overlays/log_signer/Dockerfile .
+          docker build -t quay.io/securesign/trillian_log_signer:${TRILLIAN_VER} .
+          docker push quay.io/securesign/trillian_log_signer:${TRILLIAN_VER}
```
Comment (lance) [inline .github/workflows/image-factory.yaml]: I think you could simplify this by providing the paths (and you should probably use podman here).
```suggestion
          podman build -t quay.io/securesign/trillian_log_server:${TRILLIAN_VER} ./redhat/overlays/log_server/Dockerfile
          podman push quay.io/securesign/trillian_log_server:${TRILLIAN_VER}
          podman build -t quay.io/securesign/trillian_log_signer:${TRILLIAN_VER} ./redhat/overlays/log_signer/Dockerfile
          podman push quay.io/securesign/trillian_log_signer:${TRILLIAN_VER}
```

Comment (tommyd450) [reply .github/workflows/image-factory.yaml]: So the rationale behind the docker files being moved in and out of the folder as simply because Dockerfiles are supposed to be run from root. Something similar can be seen on line 235 for the Trillian DB image build. I can fully configure them as above but it would require those Dockerfile pathings to be altered.

Comment (lance) [reply .github/workflows/image-factory.yaml]: Oh you are right. The `PATH` parameter is meant to be the root of the build, not the path to the Dockerfile. My mistake. But please do use podman here.

Comment (tommyd450) [reply .github/workflows/image-factory.yaml]: Will do.

**Review by lance — APPROVED** [review]
/lgtm

**Comment by lance** [comment]
Merging in spite of CI. It's failing due to inaccessible secrets in the github action. We need to split this workflow into 1) build and 2) publish, and only run publish on merge to main.
